### PR TITLE
Collected all pieces

### PR DESCRIPTION
**Pull request by ella-blaupause**

### CODE REVIEW

File: components/CollectingArea/CollectingArea.test.js [hunk 1]
```diff
@@ -1,28 +1,48 @@
 import { render, screen } from "@testing-library/react";
 import CollectingArea from ".";
 
-// eslint-disable-next-line jest/valid-title
-test("Test if the header `Eingesammelte Puzzleteile` is rendered", async () => {
-  render(<CollectingArea countPieces={0} puzzlePieces={[]} />);
-  const header = await screen.findByText(/Eingesammelte Puzzleteile/i);
+test("if the component displays the header 'Eingesammelte Puzzleteile' correctly.", async () => {
```
Comment (JessicaLoers) [inline components/CollectingArea/CollectingArea.test.js:4]: same as above, but just an suggestion 👍

```suggestion
test("displays the header with title 'Eingesammelte Puzzleteile'.", async () => {
```

File: components/CollectingArea/CollectingArea.test.js [hunk 2]
```diff
@@ -1,28 +1,48 @@
...
+  });
+
   expect(header).toBeInTheDocument();
 });
 
-// eslint-disable-next-line jest/valid-title
-test("Test that the component handles different count of collected puzzle pieces.", async () => {
-  const countPieces = 0;
-  const puzzlePieces = [{ id: 1 }, { id: 2 }, { id: 3 }];
+test("that the component handles different count of collected puzzle pieces.", async () => {
```
Comment (JessicaLoers) [inline components/CollectingArea/CollectingArea.test.js:21]: same here 
```suggestion
test("handles different count of collected puzzle pieces.", async () => {
```

**Review by JessicaLoers — CHANGES_REQUESTED** [review]
**Hi @ella-blaupause ✨**

I took a closer look not only at the PR but also at your whole project.🤓
Your structure is very clear and your code is super clean. **Really great job.**

I have left you some picky notes and comments. 
But especially the question regarding the `useEffect() `interests me very much.

File: components/Clue/Clue.test.js [hunk 3]
```diff
@@ -3,25 +3,24 @@ import Clue from ".";
 
 // eslint-disable-next-line jest/valid-title
 test("Test if each StyledColorDiv component displays the correct symbol.", async () => {
```
[the remark this answers is not in the document]
Comment (ella-blaupause) [reply components/Clue/Clue.test.js]: ✅

File: components/Clue/Clue.test.js [hunk 4]
```diff
@@ -3,25 +3,24 @@ import Clue from ".";
...
   expect(firstSymbol).toBeInTheDocument();
   expect(secondSymbol).toBeInTheDocument();
-  expect(thirdSymbol).toBeInTheDocument();
 });
 
-// eslint-disable-next-line jest/valid-title
-test("Test if each StyledColorDiv component is correctly styled with the provided color.", async () => {
-  const randomColor = ["red", "green", "blue"];
-  const randomSymbol = ["A", "B", "C"];
+test("if each StyledColorDiv component is correctly styled with the provided color.", async () => {
```
[the remark this answers is not in the document]
Comment (ella-blaupause) [reply components/Clue/Clue.test.js]: ✅

File: pages/_app.js [hunk 5]
```diff
@@ -27,7 +29,27 @@ export default function App({ Component, pageProps }) {
...
+      setCountPieces(countPieces + 1);
+      setPuzzlePieces(
+        puzzlePieces.map((puzzlePiece) => {
+          return puzzlePiece.id === puzzlePieceId
+            ? { ...puzzlePiece, isCollected: true, isCountable: false }
+            : puzzlePiece;
+        })
+      );
+    }
+  }
```
[the remark this answers is not in the document]
Comment (ella-blaupause) [reply pages/_app.js]: ✅

File: components/CollectingArea/CollectingArea.test.js (hunk 1 above)
Comment (ella-blaupause) [reply components/CollectingArea/CollectingArea.test.js]: ✅

File: components/CollectingArea/CollectingArea.test.js (hunk 2 above)
Comment (ella-blaupause) [reply components/CollectingArea/CollectingArea.test.js]: ✅

**Comment by ella-blaupause** [comment]
Thanks for your suggestions Jesica 👍
Hope I have implemented all

**Pull request merged**